src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
-     *
-     * @param selectedLine selected line number
-     * @param psiMethod surrounding PSI method
-     * @param psiFile containing PSI file
-     * @return true if the line is valid, false otherwise
-     */
-    private fun validateLine(
-        selectedLine: Int,
-        psiMethod: PsiMethod,
-        psiFile: PsiFile,
-    ): Boolean {
-        val doc: Document = PsiDocumentManager.getInstance(psiFile.project).getDocument(psiFile) ?: return false
+class JavaPsiHelper(private val psiFile: PsiFile) : PsiHelper {
 
-        val psiMethodBody: PsiCodeBlock = psiMethod.body ?: return false
-        if (psiMethodBody.statements.isEmpty()) return false
+    override val language: String
+        get() = "Java"
 
-        val firstStatement: PsiStatement = psiMethodBody.statements.first()
-        val lastStatement: PsiStatement = psiMethodBody.statements.last()
+    private val log = Logger.getInstance(this::class.java)
 
-        val firstStatementLine: Int = doc.getLineNumber(firstStatement.startOffset)
-        val lastStatementLine: Int = doc.getLineNumber(lastStatement.endOffset)
+    override fun generateMethodDescriptor(
+        psiMethod: PsiMethodWrapper,
+    ): String {
+        val methodDescriptor = psiMethod.methodDescriptor
+        log.info("Method description: $methodDescriptor")
+        return methodDescriptor
+    }
 
-        return (selectedLine in firstStatementLine..lastStatementLine)
+    override fun getSurroundingClass(
+        caretOffset: Int,
+    ): PsiClassWrapper? {
+        val classElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiClass::class.java)
+        for (cls in classElements) {
+            if (withinElement(cls, caretOffset)) {
+                val javaClassWrapper = JavaPsiClassWrapper(cls)
+                if (javaClassWrapper.isValid()) {
+                    log.info("Surrounding class for caret in $caretOffset is ${javaClassWrapper.qualifiedName}")
+                    return javaClassWrapper
+                }
+            }
+        }
+        log.info("No surrounding class for caret in $caretOffset")
+        return null
     }
 
-    /**
-     * Determines whether a given PsiElement contains a specified caret offset.
-     *
-     * @param psiElement The PsiElement to check if it contains the caret offset.
-     * @param caretOffset The caret offset to check.
-     * @return `true` if the PsiElement contains the caret offset, `false` otherwise.
-     */
-    private fun withinElement(
-        psiElement: PsiElement,
+    override fun getSurroundingMethod(
         caretOffset: Int,
-    ): Boolean {
-        return (psiElement.startOffset <= caretOffset) && (psiElement.endOffset >= caretOffset)
+    ): PsiMethodWrapper? {
+        val methodElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiMethod::class.java)
+        for (method in methodElements) {
+            if (method.body != null && withinElement(method, caretOffset)) {
+                val surroundingClass =
```

Comment:
**[neatpick]** I think the method `withinElement` would look nicer being implemented as an extension function of `PsiMethod`.